src/nu/xom/tests/CITestUtil.java

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2024 Elliotte Rusty Harold
+   
+   This library is free software; you can redistribute it and/or modify
+   it under the terms of version 2.1 of the GNU Lesser General Public 
+   License as published by the Free Software Foundation.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+   GNU Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library. If not, see
+   <https://www.gnu.org/licenses/>.
+   
+   You can contact Elliotte Rusty Harold by sending e-mail to
+   elharo@ibiblio.org. Please include the word "XOM" in the
+   subject line. The XOM home page is located at https://xom.nu/
+*/
+
+package nu.xom.tests;
+
+/**
+ * <p>
+ * Utility methods for CI-specific test handling, particularly for network-dependent tests.
+ * </p>
+ * 
+ * @author Elliotte Rusty Harold
+ * @version 1.3.9
+ *
+ */
+class CITestUtil {
+
+    // Prevent instantiation
+    private CITestUtil() {}
+    
+    /**
+     * Helper method to check if we're running in a CI environment.
+     * Network-dependent tests can use this to skip execution in CI
+     * where network connectivity may be unreliable.
+     * 
+     * @return true if running in CI, false otherwise
+     */
+    static boolean isRunningInCI() {
+        String ci = System.getenv("CI");
+        String githubActions = System.getenv("GITHUB_ACTIONS");
+        return "true".equalsIgnoreCase(ci) || "true".equalsIgnoreCase(githubActions);
+    }
+    
+    /**
+     * Helper method to check if an exception is network-related.
+     * Checks the exception chain for UnknownHostException or ConnectException.
+     * 
+     * @param ex the exception to check
+     * @return true if the exception is network-related, false otherwise
+     */
+    static boolean isNetworkException(Exception ex) {
```

Comment:
So what we really want here is to check both the type of the exception and whether we're running in a CI. We ignore the failure only if both conditions are met. 

Probably cleanest to add a third method like shouldIgnore that checks both of these